internal/controllers/topology/cluster/scope/upgradetracker.go

```diff
@@ -181,3 +192,9 @@ func (m *MachineDeploymentUpgradeTracker) DeferredUpgradeNames() []string {
 func (m *MachineDeploymentUpgradeTracker) DeferredUpgrade() bool {
 	return len(m.deferredNames) != 0
 }
+
+// HoldReconcile returns true if the MachineDeployment reconciliation needs to be held back.
+// Example: MachineDeployment reconciliation needs to be held back if it is pending an upgrade.
+func (m *MachineDeploymentUpgradeTracker) HoldReconcile(name string) bool {
+	return m.pendingNames.Has(name)
```

Comment:
q: should this be
```suggestion
	return m.pendingUpgradeNames.Has(name)
```

Reply:
This is from old code of the PR. `HoldReconcile` is dropped in the updated version of the PR.